### PR TITLE
normalise `bun add` package specifiers

### LINKED ISSUE #1920: `bun add foo` adds the dependency as `npm:foo` instead of "latest"
This is a regression from #1895 

```bash
❯ bun add iconv-lite
bun add v0.5.2 (f8f989f6)

 installed iconv-lite@0.6.3
```

package.json:
```json
{
  "name": "body-parser-test",
  "version": "1.0.0",
  "dependencies": {
    "iconv-lite": "npm:iconv-lite"
  }
}
```

It should be:

```json
{
  "name": "body-parser-test",
  "version": "1.0.0",
  "dependencies": {
    "iconv-lite": "^0.6.3"
  }
}
```

### PR DESCRIPTION
**Pull request by alexlamsl**
- fix invalid buffer references

fixes #1920

### CODE REVIEW

**Comment by Jarred-Sumner** [comment]
The title of the issue I made was wrong. It shouldn't be the literal "latest" unless the user inputs the tag "latest". Not specifying a version should make it insert  "^resolved-latest-version"

File: src/install/install.zig [hunk 1]
```diff
@@ -4305,11 +4310,32 @@ pub const PackageManager = struct {
...
+
                         if (new_dependencies[k].key == null) {
                             new_dependencies[k].key = JSAst.Expr.init(
                                 JSAst.E.String,
                                 JSAst.E.String{
-                                    .data = update.name,
+                                    .data = if (update.resolved_name.isEmpty())
+                                        update.name
+                                    else
+                                        try allocator.dupe(u8, update.resolved_name.slice(update.version_buf)),
```
Comment (alexlamsl) [inline src/install/install.zig:4338]: This will give me gibberish if I don't `.dupe()` it...

File: src/install/install.zig [hunk 2]
```diff
@@ -4400,13 +4414,16 @@ pub const PackageManager = struct {
             }
 
             for (updates) |*update| {
-                var str = update.e_string.?;
-
-                if (update.version.tag == .uninitialized) {
-                    str.data = latest;
-                } else {
-                    str.data = update.version.literal.slice(update.version_buf);
```
Comment (alexlamsl) [inline src/install/install.zig:4408]: ... however this seems to survive just fine without any extra work? :thinking:

File: src/install/install.zig (hunk 1 above)
Comment (Jarred-Sumner) [reply src/install/install.zig:4338]: E.String is recycled memory

**Comment by Jarred-Sumner** [comment]
Thanks. I'm not 100% sure the E.String part is correct - we may want to dupe that pointer as well.

**Pull request merged**